### PR TITLE
Feat: associate attributes to tracim user profile

### PR DESCRIPTION
**Pull request by Millefeuille42**
#6248 
This PR comes from https://github.com/tracim/tracim/pull/6237 so they need to be merged in order, ideally

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [ ] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [ ] The code is clear enough
- [ ] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [ ] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

### CODE REVIEW

**Review by PhilippeAccorsi — COMMENTED** [review]
ok for me

**Comment by lebouquetin** [comment]
@Millefeuille42 is it possible to use a comment-proof json file? If the parser is not our responsability, then a "complete" documentation is required.

By complete, I mean at least a markdown document embedding a json example with explanation about fields usage and allowed values.

Another solution is to add a config.json.documented with inline comments with end of lines like `// this field xxx`

I prefer the first documentation style (with a markdown) but both are ok for now.

**Comment by Millefeuille42** [comment]
@lebouquetin you can find the documentation in markdown format in `/doc/settings.md` in the SAML section. There is an example JSON provided with in depth details below it.

File: doc/backend/setting.md [hunk 1]
```diff
@@ -302,15 +312,20 @@ Additional fields specific to tracim can be found in the `virtual_organization`
...
-- metadata_cache_duration: Cache duration for remote metadata. In this example, the default cache duration is set to 86,400 seconds (1 day). 
+- metadata_cache_duration: Cache duration for remote metadata. In this example, the default cache duration is set to 86,400 seconds (1 day).
 - `virtual_organization`: Configuration for virtual organizations associated with IdPs. This is where you will define per-IdP settings. Each IdP is identified by its metadata URL.
   - `common_identifier`: A common identifier for the virtual organization associated with the IdP.
   - `logo_url`: URL to the organization's logo on the selection screen.
   - `displayed_name`: The displayed name of the organization on the selection screen.
   - `attribute_map`: Mapping of SAML attributes to specific names used within the SP.
+  - `profile_map`: Mapping of SAML attributes to tracim's user profile.
+    - `value`: Mapping of SAML attributes to the value that will be tested.
+    - `match`: Regex pattern that will be tested against value, if there is a match profile is set.
```
Comment (lebouquetin) [inline doc/backend/setting.md:323]: waht syntax is used? is it real regex processed with python `re` module? If so, I suggest to explain it

Comment (lebouquetin) [reply doc/backend/setting.md:323]: other question: when talking about this development, we talked about field combination. Is it working like this? If so, an example would be much appreciated

Comment (Millefeuille42) [reply doc/backend/setting.md:323]: It is regex processed with the python `re` module. I will make it clearer

**Comment by lebouquetin** [comment]
Ready to merge, juste waiting for a few words of documentation about the compatibility of the regex usage (based on python `re` module)

File: doc/backend/setting.md (hunk 1 above)
Comment (Millefeuille42) [reply doc/backend/setting.md:323]: @lebouquetin I'm not sure to understand your second question

Comment (lebouquetin) [reply doc/backend/setting.md:323]: is it possible to have a configuration which implements `public_name = "{firstName} {lastName}"`?

Comment (Millefeuille42) [reply doc/backend/setting.md:323]: Yes it is possible.

**Comment by Millefeuille42** [comment]
Did the changes in documentation

**Pull request merged**